File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/ClazzLog.java
package com.ustadmobile.lib.db.entities;

import com.ustadmobile.lib.database.annotation.UmEntity;
import com.ustadmobile.lib.database.annotation.UmPrimaryKey;

/**
 * Represents one session (e.g. day) in the class log book. This is related to attendance records, but
 * could also be related to behavior logs etc. in the future.
 */
@UmEntity
public class ClazzLog implements SyncableEntity{

    @UmPrimaryKey(autoIncrement = true)
    private long clazzLogUid;

    private long clazzLogClazzUid;

    private long logDate;

    private long timeRecorded;

    private long masterChangeSeqNum;

    private long localChangeSeqNum;

    public long getClazzLogUid() {
        return clazzLogUid;
    }

    public void setClazzLogUid(long clazzLogUid) {
        this.clazzLogUid = clazzLogUid;
    }

    public long getClazzLogClazzUid() {
        return clazzLogClazzUid;
    }

    public void setClazzLogClazzUid(long clazzLogClazzUid) {
        this.clazzLogClazzUid = clazzLogClazzUid;
    }

    public long getLogDate() {
        return logDate;
    }

    public void setLogDate(long logDate) {
        this.logDate = logDate;
    }

    public long getTimeRecorded() {
        return timeRecorded;
    }

    public void setTimeRecorded(long timeRecorded) {
        this.timeRecorded = timeRecorded;
    }

    @Override
    public long getMasterChangeSeqNum() {
        return masterChangeSeqNum;
    }

    @Override
    public void setMasterChangeSeqNum(long masterChangeSeqNum) {
        this.masterChangeSeqNum = masterChangeSeqNum;
    }

    @Override
    public long getLocalChangeSeqNum() {
        return localChangeSeqNum;
    }

    @Override
    public void setLocalChangeSeqNum(long localChangeSeqNum) {
        this.localChangeSeqNum = localChangeSeqNum;
    }
}

File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/ClazzLog.kt
package com.ustadmobile.lib.db.entities

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import com.ustadmobile.lib.database.annotation.UmEntity
import com.ustadmobile.lib.database.annotation.UmPrimaryKey

/**
 * Represents one session (e.g. day) in the class log book. This is related to attendance records, but
 * could also be related to behavior logs etc. in the future.
 */
@UmEntity
@Entity
open class ClazzLog(
    @field:UmPrimaryKey(autoIncrement = true)
    @field:PrimaryKey(autoGenerate = true)
    var clazzLogUid: Long = 0,

    var clazzLogClazzUid: Long = 0,

    var logDate: Long = 0,

    var timeRecorded: Long = 0,

    var masterChangeSeqNum: Long = 0,

    var localChangeSeqNum: Long = 0
) {

    constructor(): this(clazzLogUid = 0)
}
